// Generated from defrost_service.ifex by covesa-ifex-core
// DO NOT EDIT - regenerate with: ./generate_proto.sh
syntax = "proto3";



package swdv.defrost_service;

// Namespace: defrost
// Defrost and visibility management

// Defrost operation modes
enum defrost_mode_t {
  OFF = 0;
  WINDSHIELD_ONLY = 1;
  ALL_WINDOWS = 2;
  QUICK_CLEAR = 3;
}

// Defrostable surfaces
enum surface_t {
  FRONT_WINDSHIELD = 0;
  REAR_WINDOW = 1;
  SIDE_MIRRORS = 2;
  SIDE_WINDOWS = 3;
}

// Visibility condition
enum visibility_status_t {
  CLEAR = 0;
  LIGHT_FOG = 1;
  HEAVY_FOG = 2;
  FROST = 3;
  ICE = 4;
}

// Defrost operation request
message defrost_request_t {
  defrost_mode_t mode = 1;
  uint32 max_duration_minutes = 2;
  bool auto_stop_when_clear = 3;
}

// Status of a defrostable surface
message surface_status_t {
  surface_t surface = 1;
  visibility_status_t visibility = 2;
  bool heater_active = 3;
  float temperature_celsius = 4;
}

// Method: start_defrost
// Start defrost operation
message start_defrost_request {
  defrost_request_t request = 1;
}

message start_defrost_response {
  string operation_id = 1;
  bool accepted = 2;
  uint32 estimated_clear_time_seconds = 3;
}

service start_defrost_service {
  rpc start_defrost(start_defrost_request) returns (start_defrost_response);
}

// Method: stop_defrost
// Stop active defrost operation
message stop_defrost_request {
  string operation_id = 1;
}

message stop_defrost_response {
  bool success = 1;
}

service stop_defrost_service {
  rpc stop_defrost(stop_defrost_request) returns (stop_defrost_response);
}

// Method: get_visibility_status
// Get current visibility status
message get_visibility_status_request {
}

message get_visibility_status_response {
  repeated surface_status_t surfaces = 1;
  bool overall_visibility_safe = 2;
}

service get_visibility_status_service {
  rpc get_visibility_status(get_visibility_status_request) returns (get_visibility_status_response);
}

// Method: quick_clear_windshield
// Emergency quick clear for windshield
message quick_clear_windshield_request {
}

message quick_clear_windshield_response {
  bool activated = 1;
  uint32 estimated_clear_seconds = 2;
}

service quick_clear_windshield_service {
  rpc quick_clear_windshield(quick_clear_windshield_request) returns (quick_clear_windshield_response);
}


